src/components/base/Upload/index.tsx

```diff
@@ -0,0 +1,119 @@
+import type {
+  ChangeEvent,
+  ChangeEventHandler,
+  DragEvent,
+  ReactElement
+} from 'react';
+import { useState, useRef } from 'react';
+import { AcceptType } from './types';
+import type { UploadProps } from './types';
+import { StyledUploadContainer, StyledInput } from './styled';
+import defaultUploader from '../../../assets/defaultUploader.svg';
+
+const ZERO = 0;
+
+const Upload = ({
+  children,
+  droppable = true,
+  name = 'FileUploadInput',
+  accept = 'img',
+  value,
+  onChangeFile,
+  ...props
+}: UploadProps): ReactElement => {
+  const [file, setFile] = useState(value);
+  const [dragging, setDragging] = useState(false);
+  const [imgSrc, setImgSrc] = useState(defaultUploader);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleFileChange: ChangeEventHandler<HTMLInputElement> = (
+    e: ChangeEvent<HTMLInputElement>
+  ): void => {
+    const target = e.target as HTMLInputElement;
+    const files = target.files;
+    if (!files) return;
+
+    const changedFile: File = files[ZERO];
+    const fileUrl = URL.createObjectURL(changedFile);
+
+    setImgSrc(fileUrl);
+    setFile(changedFile);
+    onChangeFile?.(changedFile);
+  };
```

Comment:
```suggestion
  const handleFileChange = (
    e: ChangeEvent<HTMLInputElement> & DragEvent<HTMLDivElement>
  ): void => {
    let fileContainer: HTMLInputElement | DataTransfer;
    if (droppable && e.dataTransfer) {
      e.preventDefault();
      e.stopPropagation();
      fileContainer = e.dataTransfer;
    } else {
      fileContainer = e.target;
    }
    const files = fileContainer.files;
    if (!files) return;

    const changedFile: File = files[ZERO];

    if (changedFile.type.includes(AcceptType[accept].split('/')[ZERO])) {
      const fileUrl = URL.createObjectURL(changedFile);
      setImgSrc(fileUrl);
      setFile(changedFile);
      onChangeFile?.(changedFile);
    }

    if (e.dataTransfer) setDragging(false);
  };
```

하단의 handleFileDrop 함수를 해당 함수 하나로 기능을 통합시켜보았습니다. 
이렇게는 어떨까요? 이렇게되면 비슷한 기능을 수행하는 handleFileDrop 함수를 정의하지 않아도 될것같습니당.
테스트해본결과 문제는 없어보여요.